src/Features/CSharp/Portable/ExtractMethod/CSharpSelectionResult.cs

```diff
@@ -101,6 +101,64 @@ AnonymousMethodExpressionSyntax or
                    current.GetLastToken().Span.End <= lastToken.Span.End;
         }
 
+        public override bool IsExtractMethodOnSingleStatement()
+        {
+            var firstStatement = this.GetFirstStatement();
+            var lastStatement = this.GetLastStatement();
+
+            return firstStatement == lastStatement || firstStatement.Span.Contains(lastStatement.Span);
+        }
```

Comment:
these are just moves from the C# generator to the C# selection result type.

Note: i think these methods are all the same as the VB equivalents.  if so, i will make a followup pr that unifies this logic int he base type.